react-components/src/hooks/useCameraFromScene.tsx

```diff
@@ -0,0 +1,40 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import { useEffect } from 'react';
+import { useSceneConfigQuery } from './useSceneConfigQuery';
+import * as THREE from 'three';
+import { useReveal } from '..';
+
+export const useCameraFromScene = (sceneExternalId: string, sceneSpaceId: string): void => {
+  const scene = useSceneConfigQuery(sceneExternalId, sceneSpaceId);
+  const viewer = useReveal();
+
+  useEffect(() => {
+    if (scene.data === undefined || scene.data === null) {
+      return;
+    }
+
+    const position = new THREE.Vector3(
+      scene.data.sceneConfiguration.cameraTranslationX,
+      scene.data.sceneConfiguration.cameraTranslationY,
+      scene.data.sceneConfiguration.cameraTranslationZ
+    );
+
+    const euler = new THREE.Euler(
+      scene.data.sceneConfiguration.cameraEulerRotationX,
+      scene.data.sceneConfiguration.cameraEulerRotationY,
+      scene.data.sceneConfiguration.cameraEulerRotationZ,
+      'XYZ'
+    );
+
+    const quaternion = new THREE.Quaternion().setFromEuler(euler);
+    const vec = new THREE.Vector3(0, 0, 1).applyQuaternion(quaternion);
+
+    // There is a bug with setCameraState rotation, so must first set transform
+    // and then lookAt the correct direction
+    viewer.cameraManager.setCameraState({ position });
+    viewer.cameraManager.getCamera().lookAt(position.clone().add(vec));
```

Comment:
We should then investigate the issue within reveal. The reason why we always should use setCameraState is because changes in camera transformation affects when and how to render sectors of the model and we track "changes" through the method as well.

Reply:
I also just realised that it is probably better to set camera parameteres with position and target instead of rotation. What do you think about that?

Reply:
It will be also much easier to convert it from/to CDF coordinates